source/meta/kernel-4/metadata.xsd

```diff
@@ -604,6 +608,63 @@ Use the complete title of a license and include version information if applicabl
             </xs:sequence>
           </xs:complexType>
         </xs:element>
+        <xs:element name="distributions" minOccurs="0">
+          <xs:complexType>
+            <xs:sequence>
+              <xs:element name="distribution" minOccurs="1" maxOccurs="unbounded">
+                <xs:annotation>
+                  <xs:documentation>Represents an accessible form of a resource such as downloadable files. This property can be repeated if different variations are available.</xs:documentation>
+                  <xs:documentation>The use of this property indicates directly downloadable distributions. Every distribution should represent the same resource in its entirety.</xs:documentation>
+                </xs:annotation>
+                <xs:complexType>
+                  <xs:sequence>
+                    <xs:element name="file" minOccurs="1" maxOccurs="unbounded">
+                      <xs:complexType>
+                        <xs:sequence>
+                          <xs:element name="contentURL" minOccurs="1" maxOccurs="1">
+                            <xs:complexType>
+                              <xs:simpleContent>
+                                <xs:extension base="xs:anyURI">
+                                  <xs:attribute name="byteSize" type="xs:unsignedInt" use="optional" />
```

Comment:
There are some attributes in elements that would benefit with adding documentation (“xs:annotation”). For example, in “byteSize” documentation explaining that the value should be a non-negative integer could provide more clarity about the expected data type of the attribute, as it explicitly indicates that the value must be a non-negative integer, even when “xs:unsignedInt” is the correct type. That also would help developers with error messaging as it will be inside the schema.